src/plugins/path/path.c

```diff
@@ -37,16 +61,20 @@ static int validateKey (Key * key, Key * parentKey)
 	const Key * meta = keyGetMeta (key, "check/path");
 	if (stat (keyString (key), &buf) == -1)
 	{
-		char * errmsg = elektraMalloc (ERRORMSG_LENGTH + 1 + +keyGetNameSize (key) + keyGetValueSize (key) +
+		char * errmsg = elektraMalloc (ERRORMSG_LENGTH + 1 + keyGetNameSize (key) + keyGetValueSize (key) +
 					       sizeof ("name:  value:  message: "));
-		strerror_r (errno, errmsg, ERRORMSG_LENGTH);
+		if (strerror_r (errno, errmsg, ERRORMSG_LENGTH) != 0)
+		{
+			strcat (errmsg, "Unknown error");
+		}
 		strcat (errmsg, " from key: ");
 		strcat (errmsg, keyName (key));
 		strcat (errmsg, " with path: ");
 		strcat (errmsg, keyValue (key));
-		ELEKTRA_ADD_WARNING (57, parentKey, errmsg);
+		ELEKTRA_SET_ERROR (57, parentKey, errmsg);
```

Comment:
"Macroing" those errors is redundant work I think as it might be changed with #1789 anyway.
I change those which are new for this plugin since it really enhances rebasing as @sanssecours suggested.